internal/providers/aws/aws.go

```diff
@@ -33,15 +36,21 @@ import (
 )
 
 var (
-	userdataUrl = url.URL{
+	userdataURL = url.URL{
 		Scheme: "http",
 		Host:   "169.254.169.254",
 		Path:   "2019-10-01/user-data",
 	}
+	userdataLatestURL = url.URL{
```

Comment:
We're only using this in one place, to append `api/token`.  So let's make this `imdsTokenURL` with `api/token` hardcoded in, then use it directly from `fetchAWSIMDSV2Token()`, dropping that function's `u` argument.  There's no reason that function needs to allow the caller to select the target URL.

Reply:
Fixed